gpt4all-bindings/python/gpt4all/_pyllmodel.py

```diff
@@ -28,6 +28,25 @@
 EmbeddingsType = TypeVar('EmbeddingsType', bound='list[Any]')
 
 
+# Find CUDA libraries from the official packages
+if platform.system() in ('Linux', 'Windows'):
+    try:
+        from nvidia import cuda_runtime, cublas
+    except ImportError:
+        pass  # CUDA is optional
+    else:
+        if platform.system() == 'Linux':
+            cudalib   = 'lib/libcudart.so.12'
```

Comment:
Shouldn't you test to see if they are actually installed?

Reply:
With the most recent commit, CUDA is now an optional dependency. The user will be explicitly warned if the CUDA backend failed to load and they do not have `gpt4all[cuda]` installed.